lib/nxt_http_client/client.rb

```diff
@@ -74,10 +82,22 @@ def build_headers(opts)
       opts = config.request_options.with_indifferent_access.deep_merge(opts.with_indifferent_access)
       opts[:headers] ||= {}
 
-      if config.x_request_id_proc
-        opts[:headers][XRequestId] ||= config.x_request_id_proc.call
+      opts[:headers]['Content-Type'] ||= ApplicationJson if config.json_request
+      opts[:headers]['Accept'] ||= ApplicationJson if config.json_response
+
+      if config.basic_auth
+        begin
+          config.basic_auth => { username:, password: }
```

Comment:
Hold on, what's this syntax? Is it something new? (I'm talking about the `=>`)

Reply:
Oh, it's rightward assignment.

Reply:
Yeah. This specific usage is colloquially called "rightward assignment", but it's an offshoot of Ruby 3's pattern matching (good explainer [here](https://rubyreferences.github.io/rubychanges/3.0.html#one-line-pattern-matching-with-)). You can use it for rightward assignment, if you will (`5 => x`), but more practically, for hash destructuring:

```rb
config.basic_auth => { username:, password: }

# same as
username = config.basic_auth[:username]
password = config.basic_auth[:password]
```

Reply:
You learn something every day :)